src/main/java/software/amazon/nio/spi/s3/S3FileSystemProvider.java

```diff
@@ -274,6 +274,7 @@ protected DirectoryStream<Path> newDirectoryStream(S3AsyncClient s3Client, Path
                     .contents()
                     .stream()
                     .map(s3Object -> truncateByPrefix(fs, prefix, s3Object))
```

Comment:
If we add `.delimiter("/")` to the API call then we probably don't need this map or the `truncateByPrefix` private method?

Reply:
Probably we need the map because we want to return S3Path objects, but yes, I think truncateByPrefix() won't be needed.